packages/stark-ui/src/modules/session-ui/effects/session-timeout-warning.effect.spec.ts

```diff
@@ -0,0 +1,185 @@
+
+			(<Spy>mockDialogService.open).and.returnValue({
+				afterClosed: () => {
+					return afterClosed$;
+				}
+			});
+
+			const mockObserver: Observer<any> = createSpyObj<Observer<any>>("observerSpy", ["next", "error", "complete"]);
+			const subject: ReplaySubject<any> = new ReplaySubject(1);
+			actions = subject.asObservable();
+
+			(<Spy>mockDialogService.open).calls.reset();
+
+			component.starkSessionTimeoutWarning$().subscribe(mockObserver);
+
+			subject.next(new StarkSessionTimeoutCountdownStart(20));
+
+			expect(mockSessionService.pauseUserActivityTracking).toHaveBeenCalled();
+			expect(mockObserver.next).toHaveBeenCalledTimes(1);
+			expect(mockObserver.error).not.toHaveBeenCalled();
+			expect(mockObserver.complete).not.toHaveBeenCalled();
+			expect(mockDialogService.open).toHaveBeenCalledTimes(1);
+			expect(mockDialogService.open).toHaveBeenCalledWith(StarkSessionTimeoutWarningDialogComponent, { data: 20 });
+
+			expect(mockSessionService.resumeUserActivityTracking).not.toHaveBeenCalled();
+
+			afterClosed$.next(afterClosedResult);
+
+			expect(mockSessionService.resumeUserActivityTracking).toHaveBeenCalledTimes(1);
+		});
+	});
+
+	describe("On StarkSessionTimeoutWarningClose$", () => {
+		it("Should close the dialog when the countdown finishes", () => {
+			const mockObserver: Observer<any> = createSpyObj<Observer<any>>("observerSpy", ["next", "error", "complete"]);
+
+			const subject: ReplaySubject<any> = new ReplaySubject(1);
+			actions = subject.asObservable();
+
+			(<Spy>mockDialogService.open).and.callThrough(); //simulates an open dialog
+
+			component.starkSessionTimeoutWarningClose$().subscribe(mockObserver);
+
+			subject.next(new StarkSessionTimeoutCountdownFinish());
+
+			expect(mockObserver.next).toHaveBeenCalledTimes(1);
+			expect(mockObserver.error).not.toHaveBeenCalled();
+			expect(mockObserver.complete).not.toHaveBeenCalled();
+			expect(mockDialogService.closeAll).toHaveBeenCalledTimes(1);
+		});
+	});
+
+	describe("On ngrxOnRunEffects", () => {
+		it("Shouldn't use the effects", () => {
+			const mockObserver: Observer<any> = createSpyObj<Observer<any>>("observerSpy", ["next", "error", "complete"]);
+
+			const actions$: ReplaySubject<any> = new ReplaySubject(1);
+			actions = actions$.asObservable();
+			spyOn(component.actions$, "pipe").and.callThrough();
+
+			mockSessionUiConfig.enableTimeoutWarningDialogEffects = false;
+
+			const mockResolvedEffectsSubject: Subject<any> = new Subject<any>();
+			const mockResolvedEffects$: Observable<any> = mockResolvedEffectsSubject.asObservable();
+
+			const resolvedEffectsObservable: Observable<EffectNotification> = component.ngrxOnRunEffects(mockResolvedEffects$);
+			expect(component.actions$.pipe).toHaveBeenCalledTimes(1);
+
+			resolvedEffectsObservable.subscribe(mockObserver);
+
+			(<Spy>mockObserver.next).calls.reset();
+			// stop the effects
+			actions$.next(new StarkInitializeSession(<any>{})); // stopping action
```

Comment:
- Can you remove the comment `// stopping action`? because it is no longer valid
- Can you replace this line by these so that it is clear that it doesn't matter which action is passed the Effects will be stopped anyways?:

```
actions$.next("dummy acton");
actions$.next("another dummy acton");
```